framework/src/modules/chain/dpos/account.js

```diff
@@ -0,0 +1,263 @@
+const BigNum = require('@liskhq/bignum');
+
+const _mergeRewardsAndDelegates = (delegatePks, rewards) =>
+	delegatePks.reduce((arr, delegatePk, index) => {
+		let delegate = arr.find(d => d.publicKey === delegatePk);
+
+		if (!delegate) {
+			delegate = {
+				publicKey: delegatePk,
+				reward: new BigNum(0),
+				blockCount: 0,
+			};
+			arr.push(delegate);
+		}
+
+		delegate.reward = delegate.reward.plus(rewards[index]);
+		delegate.blockCount += 1;
+		delegate.isGettingRemainingFees = index === delegatePks.length - 1;
+
+		return arr;
+	}, []);
+
+class Account {
+	constructor({
+		storage,
+		slots,
+		activeDelegates,
+		logger,
+		delegates,
+		exceptions,
+	}) {
+		this.storage = storage;
+		this.slots = slots;
+		this.activeDelegates = activeDelegates;
+		this.logger = logger;
+		this.delegates = delegates;
+		this.exceptions = exceptions;
+	}
+
+	async apply(block, tx) {
+		return this._update(block, false, tx);
+	}
+
+	async undo(block, tx) {
+		return this._update(block, true, tx);
+	}
+
+	/**
+	 * @param {Block} block
+	 */
+	async _update(block, undo, tx) {
+		// @todo add proper description
+		if (block.height === 1) {
+			return false;
+		}
+
+		await this._updateProducedBlocks(block, undo, tx);
+
+		if (this._isLastBlockOfTheRound(block)) {
+			const roundSummary = await this._summarizeRound(block, tx);
+
+			const nonForgedDelegatePks = await this._getNonForgedDelegatePublicKeys(
```

Comment:
`nonForgedDelegatePks` I'm not sure what the correct name of this variable should be but the current one doesn't convey much information about what's going on. Is this a list of Public Keys of delegates that did not forge any block in this round? If so maybe `noBlocksProducedDelegatesPKs` could be better?